content/docs/user-guide/assets/scene-pipeline/scene-graph.md

```diff
@@ -0,0 +1,170 @@
+
+## Notable features
+
+### Name indexing
+
+Nodes in the ```SceneGraph``` have a name that can be used to locate it at a later time. Names don't need to be unique although a direct sibling cannot have the same name. When a new node is added its full name is composed as a list of its parent, starting at the root, separated by a dot. As an example, a node with the name "Child" that has a parent called "Parent" would have the full unique name "Parent.Child". The stored name object returns the full path by calling ```GetPath```, while only the name portion can be returned with ```GetName```.
+
+### End points
+
+Nodes can be marked as "end points". An end point node will not accept children and throws an assert when attempted. The Scene pipeline uses this to mark nodes as being an attribute, for instance. This allows distinguishing a mesh transform from a group transform for instance as they both have the same data, but one is an attribute where the other isn't. It also makes it easier to iterate over only the attributes of a node and skip over the children that form new groups. What end points exactly means will depend on how the ```SceneGraph``` is used, but it always guarantees that end point nodes will not have any children.
+
+### Hierarchical and flat retrieval
+
+While the core functionality of the ```SceneGraph``` is to provide hierarchical information about nodes, this information is not always needed. Sometimes all that's needed is a list of the names, or all objects of a certain type. If a tool in the editor for instance needs to present a drop-down list of all bones in a scene, only the names and the content is needed and iterating through them as a flat list is more efficient, yet when the selected bone is exported, the hierarchical information will be needed to find that bones children and attributes. The ```SceneGraph``` provides optimal support for both situations, using one or several approaches as needed (more on this later).
+
+{{< note >}}
+Note: while it can be tempting to use the name in some situations to look up the parent, this is both more expensive than a hierarchical lookup and not guaranteed to work in the future.
+{{< /note >}}
+
+### Guaranteed root node
+
+The ```SceneGraph``` is guaranteed to have at least one node. By default, this is a nameless node at the root. This guarantee avoids additional complexities having to deal with the edge case of no nodes. This both helps reduce the API for the ```SceneGraph``` and makes it easier to work with as there's no need for any tests to deal with having no entries. It is something though that has to be kept in mind if a blank root node is not an option. This root node behaves in the same way as other nodes so data and siblings can be assigned to it.
+
+## Internal layout
+
+{{< note >}}
+ADD GRAPH
```

Comment:
This shouldn't get checked in. I think bringing in these docs changes in two steps, 1 without images and a second pull request that adds images will be fine.

Reply:
added